osu.Game.Rulesets.Taiko/Mods/TaikoInputBlockingMod.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Game.Beatmaps.Timing;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
+using osu.Game.Screens.Play;
+using osu.Game.Utils;
+using osu.Game.Rulesets.Taiko.UI;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public abstract partial class TaikoInputBlockingMod : Mod, IApplicableToDrawableRuleset<TaikoHitObject>, IUpdatableByPlayfield
```

Comment:
I'm skeptical of this structure. For the initial implementation I would not do `TaikoInputBlockingMod` at all, but implement the single tap mod in a single file. No inheritance (except from `Mod`, obviously).

The rationale is that I don't know what the alternate mod will look like, and I have no way of knowing if the structure put down here is going to nicely extend to mod. In fact I have major doubts it will; `LastAccepted{Don,Kat}Action` likely have no place in this base class once people start with their kkdd/kddk/ddkk alternating madness.

Reply:
Sounds reasonable, I'll get around to it